### DISCUSSION TITLE
@mobily/ts-belt for inspiration

### DISCUSSION THREAD

**Discussion by ivan-kleshnin** [selfrefactor/rambda#662] [Ideas]
There's a new library `@mobily/ts-belt` which you can check for reference, Dejan. You are likely aware of them (I see a single mention) but anyway.

Not a fan of Rescript... but I've noticed a lot of bold, non-conventional decisions, regarding APIs. 

For example, their `map` accepts either `(mapFn)(data)` or `(data, mapFn)` 👈  the flipped order. Finicky but might be convenient in practice: 

```ts
export declare function map<A, B>(xs: Array<A>, mapFn: (_1: A) => B): Array<B>

export declare function map<A, B>(
  mapFn: (_1: A) => B,
): (xs: Array<A>) => Array<B>
```

More important is working type inference for `compose / pipe`, unlike Ramda/Rambda:

![Screen Shot 2022-12-04 at 12 44 38](https://user-images.githubusercontent.com/2128182/205484286-2bec8dfc-c8f5-4c07-8593-8e3faa012724.png)

![Screen Shot 2022-12-04 at 12 44 57](https://user-images.githubusercontent.com/2128182/205484292-975ff3fa-c1eb-4162-9865-d778acd76a80.png)

👆 `pipe` accepts `data` as the first argument which is again non-conventional but improves readability (and maybe type inference).

**Comment by selfrefactor** [comment]
Good reference, but I will use as reference for `Roza` library. As far as Rambda is concerned, I intend to stop looking that much for new solutions but rather simply keep up with `@types/ramda`. I do so by checking regularly the changes there.

**Comment by Peims** [comment]
Thanks for your  email, I will reply as soon as I see it. -----Maosong Pei
